### PR TITLE
Database directory created

### PR DESCRIPTION
**Pull request by geekanamika**
Please refer to design doc for all information : https://docs.google.com/document/d/1tvuckcfnPufGBVpbZC0wizlSwg203O2n5i6tDKw9KfY/edit#

### CODE REVIEW

File: malaria-app-android/src/main/java/com/peacecorps/malaria/ui/user_medicine_setting/UserMedicineSettingsFragmentActivity.java [hunk 1]
```diff
@@ -26,35 +26,49 @@
...
     private static String mDrugPicked;
     private static int mHour;
     private static int mMinute;
     private final static Calendar mCalendar = Calendar.getInstance();
     private String TAGUMSFA = "UserMedicineSettingsFragmentActivity";
-    static SharedPreferenceStore mSharedPreferenceStore;
-    private static View v;
-    private static TimePicker tp;
 
+    static SharedPreferenceStore mSharedPreferenceStore;
```
[the remark this answers is not in the document]
Comment (geekanamika) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/ui/user_medicine_setting/UserMedicineSettingsFragmentActivity.java]: I'll handle it in next PR when completing activity's implementation.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    {
+        Date dobj=null;
+
+        SimpleDateFormat sdf=new SimpleDateFormat("yyyy-MM-dd");
+        try {
+            dobj= sdf.parse(s);
+        }
+        catch (ParseException e)
+        {
+            e.printStackTrace();
```
Comment (manjumegh) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java:53]: Please, takecare of exception formating.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    /*Getting the Day of Week from the String**/
+    public static int getDayofWeek(Date d)
+    {
+        Calendar cal=Calendar.getInstance();
+        cal.setTime(d);
+        int day=cal.get(Calendar.DAY_OF_WEEK);
+        return day;
+    }
+    /*Getting no. of Days between two interval**/
+    public static long getNumberOfDays(Date d1,Date d2) {
```
Comment (manjumegh) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java:68]: I would suggest to add @nonnull, or, check for d1, or/and d2 null, how to handle.

Comment (manjumegh) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java:68]: make to take care of null check, in all the methods, as I don't want to comment on each method for this.

**Review by manjumegh — APPROVED** [review]
Looks good,  As you mentioned, you have already takencare of codacy issue.Just fix minor comment.

**Review by niksj — APPROVED** [review]
Looks good.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java (hunk 2 above)
Comment (geekanamika) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/utils/CalendarFunction.java:53]: Updating it in next PR :+1:

**Pull request merged**